### PR TITLE
Amazon linux

### PR DESCRIPTION
**Pull request by samh241**
Adding in workflow for amazon linux 2023

### CODE REVIEW

**Comment by geerlingguy** [comment]
You may be able to just add it under the distro list... https://github.com/geerlingguy/ansible-role-java/blob/31f43f54e2ee2dafedd8c3f6496a687208d43790/.github/workflows/ci.yml#L43

**Comment by samh241** [comment]
Yea I did add it in on my fork, but getting the same error of  the **Define java_packages tasks**: `The error was: '__java_packages' is undefined. `. Not sure what I'm missing but will have a dig today.

**Comment by geerlingguy** [comment]
Possibly this line not picking up the right vars file? https://github.com/geerlingguy/ansible-role-java/pull/143/files#diff-3d0ff1709ca48add100327bb2a468e6c508fb92a159c64c4f99ad1df89d9bddeR27

**Comment by geerlingguy** [comment]
I noticed:

```
TASK [geerlingguy.java : Include version-specific variables for Amazon.] *******
  skipping: [instance]
```

That seems like it should not be skipped if you are running Amazon Linux.

**Comment by samh241** [comment]
Ah yea I see it now 😄, thanks Jeff 👍

File: playbook.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
```
Comment (geerlingguy) [inline playbook.yml:1]: This file shouldn't be added - can you remove?

File: tasks/main.yml [hunk 2]
```diff
@@ -3,6 +3,10 @@
   include_vars: "{{ ansible_distribution }}.yml"
   when: ansible_distribution == 'FreeBSD' or ansible_distribution == 'Fedora'
 
+- name: Include OS-specific variables for Amazon.
+  include_vars: "{{ ansible_distribution }}-{{ ansible_distribution_version}}.yml"
```
Comment (geerlingguy) [inline tasks/main.yml:7]: It seems like we should either have this task, and the associated vars files, or the other task ('version-specific') and associated vars files, but not both. They seem to duplicate the effort.

**Comment by samh241** [comment]
Sorry didn't realise I hadn't cleaned up.

File: tasks/main.yml (hunk 2 above)
Comment (samh241) [reply tasks/main.yml:7]: Yea, I can see what I've done there, have updated to for the former

**Pull request merged**